daemon/main.go

```diff
@@ -769,11 +769,14 @@ func initEnv(cmd *cobra.Command) {
 func runCiliumHealthEndpoint(d *Daemon) error {
 	// PingEndpoint will always fail the first time (initialization).
 	if err := health.PingEndpoint(); err != nil {
+		localNode := node.GetLocalNode()
```

Comment:
This introduces race conditions now as localNode is not using any mutex to protect its fields. I would keep the old way of retrieving the health IPs.

Reply:
The reason for this is that we now access localNode in the context of go routines